fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java

```diff
@@ -179,12 +176,6 @@ protected void runAfterCatalogReady() {
             }
         } // end for all results
 
-        // we also add a 'mocked' master Frontend heartbeat response to synchronize master info to other Frontends.
```

Comment:
why removed?

Reply:
This is duplicated code, the leader's heartbeat response has been added at
```
        // send frontend heartbeat
        List<Frontend> frontends = GlobalStateMgr.getCurrentState().getNodeMgr().getFrontends(null);
        for (Frontend frontend : frontends) {
            FrontendHeartbeatHandler handler = new FrontendHeartbeatHandler(frontend,
                    GlobalStateMgr.getCurrentState().getNodeMgr().getClusterId(),
                    GlobalStateMgr.getCurrentState().getNodeMgr().getToken());
            hbResponses.add(executor.submit(handler));
        }
```

```
            if (fe.getHost().equals(GlobalStateMgr.getCurrentState().getNodeMgr().getSelfNode().first)) {
                // heartbeat to self
                if (GlobalStateMgr.getCurrentState().isReady()) {
                    return new FrontendHbResponse(fe.getNodeName(), Config.query_port, Config.rpc_port,
                            GlobalStateMgr.getCurrentState().getMaxJournalId(), System.currentTimeMillis(),
                            GlobalStateMgr.getCurrentState().getFeStartTime(),
                            Version.STARROCKS_VERSION + "-" + Version.STARROCKS_COMMIT_HASH,
                            JvmStats.getJvmHeapUsedPercent());
                } else {
                    return new FrontendHbResponse(fe.getNodeName(), "not ready");
                }
            }
```